drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -428,6 +433,17 @@ DEVICE_AND_API_INIT(nrf5_154_radio, CONFIG_IEEE802154_NRF5_DRV_NAME,
 		    nrf5_init, &nrf5_data, &nrf5_radio_cfg,
 		    POST_KERNEL, CONFIG_IEEE802154_NRF5_INIT_PRIO,
 		    &nrf5_radio_api);
+#elif defined(CONFIG_NET_L2_OPENTHREAD)
+NET_DEVICE_INIT(nrf5_154_radio, CONFIG_IEEE802154_NRF5_DRV_NAME,
+		nrf5_init, &nrf5_data, &nrf5_radio_cfg,
+		CONFIG_IEEE802154_NRF5_INIT_PRIO,
+		&nrf5_radio_api, OPENTHREAD_L2,
+		NET_L2_GET_CTX_TYPE(OPENTHREAD_L2), 1280);
+
+NET_STACK_INFO_ADDR(RX, nrf5_154_radio,
+		    CONFIG_IEEE802154_NRF5_RX_STACK_SIZE,
+		    CONFIG_IEEE802154_NRF5_RX_STACK_SIZE,
+		    nrf5_data.rx_stack, 0);
```

Comment:
too many lines for just 3 parameters that change.

You should be able to keep only one NET_DEVICE_INIT and one NET_STACK_INFO_ADDR calls, the 3 parameters that changes should be set via in-betweed macros and that should do it.

Reply:
I've tried this:
```c
#if defined(CONFIG_NET_L2_IEEE802154)
#define L2 IEEE802154_L2
#define MTU 125
#elif defined(CONFIG_NET_L2_OPENTHREAD)
#define L2 OPENTHREAD_L2
#define MTU 1280
#endif

#if defined(CONFIG_NET_L2_IEEE802154) || defined(CONFIG_NET_L2_OPENTHREAD)
NET_DEVICE_INIT(nrf5_154_radio, CONFIG_IEEE802154_NRF5_DRV_NAME,
		nrf5_init, &nrf5_data, &nrf5_radio_cfg,
		CONFIG_IEEE802154_NRF5_INIT_PRIO,
		&nrf5_radio_api, L2,
		NET_L2_GET_CTX_TYPE(L2), MTU);

NET_STACK_INFO_ADDR(RX, nrf5_154_radio,
		    CONFIG_IEEE802154_NRF5_RX_STACK_SIZE,
		    CONFIG_IEEE802154_NRF5_RX_STACK_SIZE,
		    nrf5_data.rx_stack, 0);
#else
DEVICE_AND_API_INIT(nrf5_154_radio, CONFIG_IEEE802154_NRF5_DRV_NAME,
		    nrf5_init, &nrf5_data, &nrf5_radio_cfg,
		    POST_KERNEL, CONFIG_IEEE802154_NRF5_INIT_PRIO,
		    &nrf5_radio_api);
#endif
```
But there is a problem with `NET_L2_GET_CTX_TYPE()` expansion as it wil expand to L2_CTX_TYPE and I cannot see a way to expand it to correct name.